Running Muxed Provider Results in Provider Schema Mismatch

## Description

When attempting to build and use this provider with development overrides, terraform-plugin-mux will return an error about the provider schemas being mismatched due to versions:

```terraform
terraform {
  required_providers {
    corner = {
      source = "hashicorp/corner"
    }
  }

  required_version = "1.0.3"
}

data "corner_time" "test" {}

output "test" {
  value = data.corner_time.test.id
}
```

```console
$ terraform apply
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - hashicorp/corner in /Users/bflad/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible
│ with published releases.
╵

Stack trace from the terraform-provider-corner plugin:

panic: got a different provider schema from two servers (*schema.GRPCProviderServer, protocol.server). Provider schemas must be identical across providers. Diff:   &tfprotov5.Schema{
- 	Version: 1,
+ 	Version: 0,
  	Block: &tfprotov5.SchemaBlock{
- 		Version:    1,
+ 		Version:    0,
  		Attributes: nil,
  		BlockTypes: nil,
  		... // 3 identical fields
  	},
  }


goroutine 1 [running]:
main.main()
	/Users/bflad/src/github.com/hashicorp/terraform-provider-corner/main.go:17 +0x2a5

Error: The terraform-provider-corner plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

╷
│ Error: Could not load plugin
│
│
│ Plugin reinitialization required. Please run "terraform init".
│
│ Plugins are external binaries that Terraform uses to access and manipulate
│ resources. The configuration provided requires plugins which can't be located,
│ don't satisfy the version constraints, or are otherwise incompatible.
│
│ Terraform automatically discovers provider requirements from your
│ configuration, including providers used in child modules. To see the
│ requirements and constraints, run "terraform providers".
│
│ failed to instantiate provider "registry.terraform.io/hashicorp/corner" to obtain schema: Unrecognized remote plugin message:
│
│ This usually means that the plugin is either invalid or simply
│ needs to be recompiled to support the latest protocol.
│
╵
```

In this provider's protocol version 5 code, it is setting the schema versions to `1`:

https://github.com/hashicorp/terraform-provider-corner/blob/0aead7ebfbc82b793562f2473fdc5cf14aec3a94/internal/protocolprovider/server.go#L61-L66

However terraform-plugin-sdk version 2 does not have a method for instantiating the response with schema versions set for the provider:

https://github.com/hashicorp/terraform-plugin-sdk/blob/5f0e47111e779cb308990590c18d0ed547f0dd37/helper/schema/grpc_provider.go#L71-L73

So they will default to Go's zero value for the type (`0`):

https://github.com/hashicorp/terraform-plugin-go/blob/a741d49d34b11ba14f042897e9737c08c888a152/tfprotov5/schema.go#L63

## Proposals

If providers should be setting version to `1`, terraform-plugin-sdk could be updated to always return provider schema versions (and their block versions) set to `1`. I'm guessing this could have unintended consequences though.

We could unset the version configuration in this provider's code, which should resolve the error.

--- a/internal/protocolprovider/server.go
+++ b/internal/protocolprovider/server.go
@@ -59,16 +59,11 @@ func (s server) StopProvider(ctx context.Context, req *tfprotov5.StopProviderReq
 func Server() tfprotov5.ProviderServer {
 	return server{
 		providerSchema: &tfprotov5.Schema{
-			Version: 1,
-			Block: &tfprotov5.SchemaBlock{
-				Version: 1,
-			},
+			Block: &tfprotov5.SchemaBlock{},
 		},
 		dataSourceSchemas: map[string]*tfprotov5.Schema{
 			"corner_time": {
-				Version: 1,
 				Block: &tfprotov5.SchemaBlock{
-					Version: 1,
 					Attributes: []*tfprotov5.SchemaAttribute{
 						{
 							Name:            "current",

--- a/internal/protocolv6provider/server.go
+++ b/internal/protocolv6provider/server.go
@@ -59,16 +59,11 @@ func (s server) StopProvider(ctx context.Context, req *tfprotov6.StopProviderReq
 func Server() tfprotov6.ProviderServer {
 	return server{
 		providerSchema: &tfprotov6.Schema{
-			Version: 1,
-			Block: &tfprotov6.SchemaBlock{
-				Version: 1,
-			},
+			Block: &tfprotov6.SchemaBlock{},
 		},
 		dataSourceSchemas: map[string]*tfprotov6.Schema{
 			"corner_v6_time": {
-				Version: 1,
 				Block: &tfprotov6.SchemaBlock{
-					Version: 1,
 					Attributes: []*tfprotov6.SchemaAttribute{
 						{
 							Name:            "current",

